feat: Added timezone as a subtitle in ScheduleFragment

dr0pdb opened the pull request:
Fixes #1974 

Changes: Added the timezone as a subtitle in ScheduleFragment. The use local time setting should be activated inside Settings.

Screenshots for the change: 

![screenshot_2018-01-23-01-17-38-27](https://user-images.githubusercontent.com/21277837/35241066-4caaadfa-ffdb-11e7-9374-6f47f3ccc355.png)

@iamareebjamal Please review.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java:
```diff
@@ -75,6 +79,15 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 
+        boolean isLocal = SharedPreferencesUtil.getBoolean(getString(R.string.timezone_mode_key), false);
+        if (isLocal) {
+            String offset = new SimpleDateFormat("ZZZZ", Locale.getDefault()).format(System.currentTimeMillis());
```
Move to DateUtils please

srv-twry replied:
Do you mean using the new Date and Time API ? I tried using them but they require API level 26 or above which is way too high.

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java:
```diff
@@ -75,6 +79,15 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         View view = super.onCreateView(inflater, container, savedInstanceState);
 
+        boolean isLocal = SharedPreferencesUtil.getBoolean(getString(R.string.timezone_mode_key), false);
+        if (isLocal) {
```
What if it is false?

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java:
```diff
@@ -246,6 +259,10 @@ public void onDestroyView() {
+        ActionBar supportActionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        if (supportActionBar != null) {
+            ((AppCompatActivity) getActivity()).getSupportActionBar().setSubtitle("");
+        }
```
Please add comment as to what this is doing

iamareebjamal replied:
Extract to a mehod -> `setSubTitle`

dr0pdb commented:
Changes: Using DateUtils, Added relevant comments and extracted common code to a method.

@iamareebjamal Please review.

Screens:
![screenshot_2018-01-28-18-54-32-91](https://user-images.githubusercontent.com/21277837/35482569-6795fc92-045d-11e8-8a19-bfe767b13d32.png)
![screenshot_2018-01-28-18-54-17-46](https://user-images.githubusercontent.com/21277837/35482571-696525a2-045d-11e8-9838-09674260e924.png)

iamareebjamal commented on android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java:
```diff
@@ -75,6 +81,12 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         View view = super.onCreateView(inflater, container, savedInstanceState);
 
+        //showing timezone with offset.
+        DateConverter.setShowLocalTime(SharedPreferencesUtil.getBoolean(getString(R.string.timezone_mode_key), false));
```
Why setting it here?

dr0pdb replied:
Just to make sure it is set, It may be redundant but i just wanted to be 100% sure.

dr0pdb commented:
Changes: Removed redundant call and moved the setting subtitle call to ```onResume``` so that it is refreshed when settings are changed in SettingsActivity

@iamareebjamal Please review.

dr0pdb committed:
feat: Add timezone as subtitle in ScheduleFragment
